mrmustard/physics/ansatz/array_ansatz.py

```diff
@@ -56,9 +56,10 @@ class ArrayAnsatz(Ansatz):
     def __init__(self, array: Batch[Tensor] | None, batch_dims: int = 0):
         super().__init__()
         self._array = array
-        self._batch_dims = batch_dims
-        self._batch_shape = tuple(self._array.shape[:batch_dims]) if array is not None else ()
         self._original_abc_data = None
+        self._batch_dims = batch_dims
+        if array is not None:
+            self._batch_shape = tuple(self._array.shape[:batch_dims])
```

Comment:
So now there is no `_batch_shape` if `array is None`, whereas before it was `()`?

Reply:
And similar in `polyexp_ansatz.py`